services/index.ts

```diff
@@ -1,38 +1,79 @@
-import * as express from 'express'
+import { graphiqlExpress, graphqlExpress } from 'apollo-server-express'
 import * as bodyParser from 'body-parser'
+import * as express from 'express'
 import { readFileSync } from 'fs'
-import { join } from 'path'
-import { graphqlExpress, graphiqlExpress } from 'apollo-server-express'
 import { makeExecutableSchema } from 'graphql-tools'
 import { merge } from 'lodash'
+import { join } from 'path'
 
-import logger from './utilities/logger'
-import emailResolvers from './handlers/email'
 import authenticationResolvers from './handlers/authentication'
+import dummyResolver from './handlers/dummy'
+import emailResolvers from './handlers/email'
+import { userPayloadFromRequest } from './utilities/auth'
+import logger from './utilities/logger'
 
+const cookieParser = require('cookie-parser')
 const typeDefs = readFileSync(join(__dirname, './schema.graphql')).toString()
-const resolvers = merge(
-    emailResolvers,
-    authenticationResolvers
-)
+const resolvers = merge(emailResolvers, authenticationResolvers, dummyResolver)
+
 const schema = makeExecutableSchema({ typeDefs, resolvers })
 
 const app = express()
 
 // TODO: should be replaced by https://github.com/hasura/graphql-engine/issues/2461
 const validateRequest = (req, res, next) => {
-    const isInvalidated = !req.headers['x-hasura-role'] || req.headers['x-hasura-role'] === 'anonymous'
-    const isIntrospection = req.body['operationName'] === 'IntrospectionQuery'
-    const isInternal = ['sendMail'].some(service => req.body['query'].includes(service))
-
-    if (isInvalidated && isInternal && !isIntrospection) {
-        res.status(403).json({
-            errors: ['Unauthorized access']
-        })
-    }
-    next()
+  if (!req.body['query']) {
+    return res.status(400).json({
+      errors: ['Document must not be empty'],
```

Comment:
I don't think we need to tell the client that a `/graphql` endpoint requires a payload.

Reply:
And if so it should have been `418`

Reply:
removed.